docs/CartRestApiDoc.md

```diff
@@ -0,0 +1,515 @@
+
+### ShoppingSession
+
+| Parameter          | Type      | Required | Description                   |
+|--------------------|-----------|----------|-------------------------------|
+| shoppingSessionId  | string    | true | The identifier of the shopping session (the cart info)   |
+| itemsAmount        | integer   | true | The amount of items          |
+| totalPrice         | number    | true | The total price of all items |
+| itemList           | item list | true | The item list                |
+
+### Item
+
+| Parameter          | Type      | Required | Description                   |
+|--------------------|-----------|----------|-------------------------------|
+| itemId             | integer   | true | The identifier of the item    |
+| productsAmount     | integer   | true | The amount of products      |
+| totalProductsPrice | integer   | true | The total products price     |
+| product            | integer   | true | The product description       |
+
+**Response Example:**
+
+```json
+{
+    "shoppingSessionId": "fddfgd-fdgdfgdfh-hdfhdfh-436346dfhd-hdfhdf",
+    "itemsAmount": "2",
+    "totalPrice": "3200",
+    "itemList": [
+        {
+            "itemId": "463463gfd-436fdhtgery3-hdhgdfghdgh46-hdhdghd",
+            "productsAmount": "2",
+            "totalProductsPrice": "1500",
+            "product": {
+                "productData": "productData"
+            }
+        },
+        {
+            "itemId": "463463gfd-436fdhtgery3-hdhgdfghdgh46-hdhdghd",
+            "productsAmount": "3",
+            "totalProductsPrice": "1700",
+            "product": {
+                "productData": "productData"
+            }
+        }
+    ]
+}
+```
+
+**400 Bad Request**
+
+If the client sends an invalid request, a 400 Bad Request should be returned.
+
+| Parameter | Type      | Required | Description                                         |
+|-----------|-----------|----------|-----------------------------------------------------|
+| message | string    | true | Error message indicating the reason for the failure. |
+| timestamp | timestamp | true | Indicating the failure time.                        |
+| code | integer   | true | Error code.    |
+
+**Response Examples:**
+
+```json
+{
+    "message": "The 'shoppingSessionId' cannot be empty.",
+    "timestamp": "2023-06-04 18:24:54",
+    "code": 12
+}
+```
+
+```json
+{
+    "message": "Invalid 'shoppingSessionId' value.",
+    "timestamp": "2023-06-04 18:24:54",
+    "code": 13
+}
+```
+
+```json
+{
+    "message": "The 'itemList' cannot be empty.",
+    "timestamp": "2023-06-04 18:24:54",
+    "code": 14
+}
+```
+
+```json
+{
+    "message": "Invalid 'itemList' value.",
+    "timestamp": "2023-06-04 18:24:54",
+    "code": 15
+}
+```
+
+--------------------------------------------------------------------------
+
+## 4. Update the amount of the specified item products in the cart
```

Comment:
please use HTTP PUT method for it.

if you update the quantity of the specified product in the cart, correct url might look as follows:

HTTP PUT .../api/v1/cart/items/{item_id}/amount

Reply:
Fixed